From 3c802e40890cb12de8b6328f127ba8b1bbb1ecd7 Mon Sep 17 00:00:00 2001
From: auto commit <unknown@unknown>
Date: Tue, 15 Jul 2008 14:57:06 -0400
Subject: [PATCH] auto_msg: importing markers-use-imv-jump-liveliness.patch

This is an automatic import of patch markers-use-imv-jump-liveliness.patch, no headers were
detected and a default message was constructed
---
 include/linux/marker.h |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/include/linux/marker.h b/include/linux/marker.h
index f49e415..af4aefb 100644
--- a/include/linux/marker.h
+++ b/include/linux/marker.h
@@ -74,10 +74,13 @@ struct marker {
 		{ __mark_empty_function, NULL}, NULL };			\
 		__mark_check_format(format, ## args);			\
 		if (!generic) {						\
-			if (unlikely(imv_cond(__mark_##name.state)))	\
+			if (unlikely(imv_cond(__mark_##name.state))) {	\
+				imv_cond_end();				\
 				(*__mark_##name.call)			\
 					(&__mark_##name, call_private,	\
 					## args);			\
+			} else						\
+				imv_cond_end();				\
 		} else {						\
 			if (unlikely(_imv_read(__mark_##name.state)))	\
 				(*__mark_##name.call)			\
-- 
1.5.5.1

